packages/react/src/Button/Button.module.css

```diff
@@ -41,70 +52,119 @@
+  border: solid var(--brand-borderWidth-thin) var(--brand-button-primary-borderColor-rest);
 }
 
 .Button--primary:not(.Button[disabled], .Button[aria-disabled='true']):hover {
-  box-shadow: var(--brand-Button-shadow-primary-hover);
+  background-color: var(--brand-button-primary-bgColor-hover);
+  border-color: var(--brand-button-primary-borderColor-hover);
 }
 
-.Button--primary.Button--disabled {
-  background-color: var(--base-color-scale-gray-4);
+.Button--primary:not(.Button[disabled], .Button[aria-disabled='true']):active,
+.Button[aria-expanded='true'] {
+  background-color: var(--brand-button-primary-bgColor-active);
+  border-color: var(--brand-button-primary-borderColor-active);
+  box-shadow: var(--brand-button-primary-shadow-active);
 }
 
 .Button--label-primary {
-  color: var(--brand-color-text-onEmphasis);
+  color: var(--brand-button-primary-fgColor-rest);
+}
+
+.Button--primary:disabled,
+.Button--primary[aria-disabled='true'],
+.Button--primary.Button--disabled {
+  background-color: var(--brand-button-primary-bgColor-disabled);
+  border-color: var(--brand-button-primary-borderColor-disabled);
 }
 
+.Button--primary:disabled .Button-arrow--disabled,
+.Button--primary[aria-disabled='true'] .Button-arrow--disabled,
+.Button--primary.Button--disabled .Button-arrow--disabled,
+.Button--primary:disabled .Button__icon-visual--disabled,
+.Button--primary[aria-disabled='true'] .Button__icon-visual--disabled,
+.Button--primary.Button--disabled .Button__icon-visual--disabled,
 .Button--label-primary.Button-label--disabled {
-  color: var(--brand-Button-fg-primary-disabled);
+  color: var(--brand-button-primary-fgColor-disabled);
 }
 
+/* secondary */
+
 .Button--secondary {
-  color: var(--brand-color-text-default);
-  box-shadow: var(--brand-Button-shadow-secondary-default);
-  transition: box-shadow 200ms;
+  transition: background-color, border-color 200ms;
+  border-color: var(--brand-button-secondary-borderColor-rest);
 }
 
 .Button--secondary:not(.Button[disabled], .Button[aria-disabled='true']):hover {
-  box-shadow: var(--brand-Button-shadow-secondary-hover);
+  background-color: var(--brand-button-secondary-bgColor-hover);
+  border-color: var(--brand-button-secondary-borderColor-hover);
+}
+
+.Button--secondary:not(.Button[disabled], .Button[aria-disabled='true']):active,
+.Button[aria-expanded='true'] {
+  background-color: var(--brand-button-secondary-bgColor-active);
+  border-color: var(--brand-button-secondary-borderColor-active);
+}
+
+.Button--label-secondary {
+  color: var(--brand-button-secondary-fgColor-rest);
+}
+
+.Button--secondary .Button__icon-visual,
+.Button--secondary .Button-arrow {
+  color: var(--brand-button-secondary-fgColor-rest);
 }
 
-.Button--label-secondary.Button-label--disabled,
-.Button--label-secondary.Button-label--disabled,
-.Button__icon-visual--disabled {
-  color: var(--brand-Button-fg-secondary-disabled);
+.Button--secondary:disabled .Button-arrow--disabled,
```

Comment:
I just need to say the `disabled` state situation with this component is absolutely bananas 🍌